### DISCUSSION TITLE
Adding a decorator for Canvas but not Docs?

### DISCUSSION THREAD

**Discussion by alexsurelee** [storybookjs/storybook#12953] [Help]
Hi there,

I'm working on a project using storybook 5 (too large to upgrade right now), and I'm trying to add `@storybook/addon-docs` to our project. Currently, we centre a lot of our components, however this doesn't play so nice with the Docs view.

Is it possible to add a decorator exclusively to the canvas view, and/or remove it exclusively for the docs view?

Cheers

**Comment by shilman** [comment]
Starting in 6.1 (which should be released soon) there's a `viewMode` that gets passed into the decorator, so you should be able to do something like:

```js
const decorator = (storyFn, { viewMode }) => {
  if(viewMode === 'story') {
    // do something
  }
  return storyFn();
}
```

**Reply by alexsurelee** [reply]
Oh man that'd be perfect, might have to try and put in the effort to get our project upgraded. If there is any v5 way however I'd still be interested in knowing!

**Reply by shilman** [reply]
```
npx sb upgrade --prerelease
```
